lib/matplotlib/font_manager.py

```diff
@@ -874,7 +874,7 @@ def json_dump(data, filename):
         try:
             json.dump(data, fh, cls=JSONEncoder, indent=2)
         except OSError as e:
-            warnings.warn('Could not save font_manager cache {}'.format(e))
+            warnings.warn('Could not save font_manager cache {}'.format(e), stacklevel=2)
```

Comment:
For example, I don't think that this warning would get clearer if using _warn_external (or even stacklevel=2).

Reply:
OK, I see what you mean.

Is it safe to assume that this warning will be "triggered" when importing pyplot? If so, then does it make sense to print the user line of code that imports pyplot i.e. ```import matplotlib.pyplot as plt```?

I tried to manipulate _warn_external to do this without success. If this solution is acceptable, any ideas how to achieve that?

Reply:
> Is it safe to assume that this warning will be "triggered" when importing pyplot?

Yes

> If so, then does it make sense to print the user line of code that imports pyplot i.e. import matplotlib.pyplot as plt?

I wouldn't overthink it (there isn't anything really smart to do in this case IMO) and would just trigger a warning at this line; or perhaps is actually makes sense to use logging.warning instead of warnings.warn here in fact, now that I think of it...

Reply:
After further thought, I convinced myself that that should exactly be the criterion for `warnings` vs `logging`: by showing the line of offending code, does that point the user to something they should modify?